woocommerce/wc-admin
# Adding Favorite API Class to Navigation Feature

Fixes #6131 

Adding the Favorites API to the navigation feature:

```php
Favorites::add_plugin($plugin );
Favorites::remove_plugin($plugin );
Favorites::get_all();
```

This simply manages an array of keys within `update_user_meta` to track the favorites for the user.

### Detailed test instructions:

-   Checkout branch
-   Run `\Automattic\WooCommerce\Admin\Features\Navigation\Favorite::add_plugin('whoa_buddy');`
-  Check `woocommerce_admin_navigation_favorites` under `wp_usermeta`, which should have content `["whoa_buddy"]`
- Run `\Automattic\WooCommerce\Admin\Features\Navigation\Favorite::add_plugin('yo_buddy');`
-  Check `woocommerce_admin_navigation_favorites` under `wp_usermeta`, which should have content `["whoa_buddy", "yo_buddy"]`
- Run `\Automattic\WooCommerce\Admin\Features\Navigation\Favorite::remove_plugin('whoa_buddy');`
-  Check `woocommerce_admin_navigation_favorites` under `wp_usermeta`, which should have content `["yo_buddy"]`
